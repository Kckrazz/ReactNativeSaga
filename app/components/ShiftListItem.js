//import liraries
import React, { Component } from "react";
import { View, StyleSheet, Dimensions } from "react-native";
import moment from "moment";
import {
  Container,
  Content,
  Card,
  CardItem,
  Text,
  Body,
  Left,
  Right,
  Thumbnail
} from "native-base";

import Ionicons from "react-native-vector-icons/Ionicons";

import constant from "../constants/ConfigConstants";

const { width } = Dimensions.get("window");

// create a component
class ShiftListItem extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }
  renderAvatar(workers) {
    var size = 3;
    return workers.slice(0, size).map(i => {
      return (
        <View style={{ paddingLeft: 10 }} key={i.id}>
          <Thumbnail small source={{ uri: constant.root.url + i.avatar }} />
        </View>
      );
    });
  }
  render() {
    const data = this.props.data;
    const start_time = moment(data.start).local().format("LT");
    const status = getShiftStatus(data.status);
    status.title = data.status ? data.status.toUpperCase() : "";
    const workers = data.workers.length > 0 ? data.workers[0] : [];
    return (
      <Card>
        <CardItem>
          <Left>
            <Body>
              <Text note>{start_time} </Text>
              <Text>{data.title}</Text>
              <Text note> {data.address}</Text>
            </Body>
          </Left>
        </CardItem>
        <CardItem>
          <Left>
            {this.renderAvatar(data.workers)}

          </Left>
          <Right>
            <View
              style={{
                flexDirection: "row",
                flex: 1,
                paddingTop: 10
              }}
            >
              <Text
                style={{
                  color: status.color,
                  fontSize: 12
                }}
              >
                {status.title}{" "}
              </Text>
              <Ionicons
                name={status.icon}
                style={{
                  fontSize: 17,
                  paddingRight: 5,
                  color: status.color
                }}
              />
            </View>
          </Right>
        </CardItem>
      </Card>
    );
  }
}

export function getShiftStatus(params) {
  switch (params) {
    case "cancelled":
      return {
        icon: "md-remove-circle",
        color: "#FF5722"
      };

    case "completed":
      return {
        icon: "md-checkmark-circle",
        color: "#2196F3"
      };

    case "approved":
      return {
        icon: "md-checkmark-circle",
        color: "#3F51B5"
      };

    case "rejected":
      return {
        icon: "md-thumbs-down",
        color: "#FF5722"
      };

    case "pending":
      return {
        icon: "md-time",
        color: "#f44336"
      };

    case "booked":
      return {
        icon: "md-calendar",
        color: "#00a65a"
      };

    case "unverified":
      return {
        icon: "md-warning",
        color: "#FF5722"
      };

    default:
      return {
        icon: "md-warning",
        color: "#FF5722"
      };
  }
}

// define your styles
const styles = StyleSheet.create({});

//make this component available to the app
export default ShiftListItem;
