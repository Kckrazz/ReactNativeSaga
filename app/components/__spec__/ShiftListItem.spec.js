import React from "react";
import "react-native";
import renderer from "react-test-renderer";
import ShiftListItem from "../ShiftListItem";
jest.unmock("../ShiftListItem");

describe("Shift List Item", () => {
  const props = {
    data: {
      id: 1,
      title: "test",
      start: "2017-05-26T13:45:00+10:00",
      end: "2017-05-26T13:45:00+10:00",
      url: "/users/areas/1/programs/1/shifts/1",
      address: "Karnataka, India",
      area: 1,
      program: 1,
      backgroundColor: "#f56954",
      borderColor: "#f56954",
      description: "asdasdasd",
      area_name: "Demo Area 2",
      clients: [],
      workers: [
        {
          id: 12,
          avatar: "placehold.it/120x120&text=image1"
        }
      ],
      status: "cancelled"
    }
  };

  test("Should Render Snapshot Correctly", () => {
    const tree = renderer.create(<ShiftListItem {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
