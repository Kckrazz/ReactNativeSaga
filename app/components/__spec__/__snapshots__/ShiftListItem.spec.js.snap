// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shift List Item Should Render Snapshot Correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderBottomWidth": 0.5,
      "borderColor": "#ccc",
      "borderLeftWidth": 0.5,
      "borderRadius": 2,
      "borderRightWidth": 0.5,
      "borderTopWidth": 0.5,
      "elevation": 3,
      "flex": 1,
      "flexWrap": "wrap",
      "marginBottom": 5,
      "marginLeft": 2,
      "marginRight": 2,
      "marginTop": 5,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 1.5,
    }
  }
  virtual={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      }
    }
    virtual={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
          "flexDirection": "row",
        }
      }
      virtual={undefined}
    >
      <View
        style={
          Object {
            "alignItems": null,
            "alignSelf": "center",
            "flex": 1,
            "marginLeft": 10,
          }
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          note={true}
          style={
            Object {
              "color": "#808080",
              "fontFamily": "System",
              "fontSize": 14,
              "fontWeight": "400",
              "marginRight": 20,
            }
          }
          virtual={undefined}
        >
          9:30 AM
           
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            }
          }
          virtual={undefined}
        >
          test
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          note={true}
          style={
            Object {
              "color": "#808080",
              "fontFamily": "System",
              "fontSize": 14,
              "fontWeight": "400",
              "marginRight": 20,
            }
          }
          virtual={undefined}
        >
           
          Karnataka, India
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      }
    }
    virtual={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
          "flexDirection": "row",
        }
      }
      virtual={undefined}
    >
      <View
        style={
          Object {
            "paddingLeft": 10,
          }
        }
      >
        <Image
          small={true}
          source={
            Object {
              "uri": "http://192.168.100.46:3000placehold.it/120x120&text=image1",
            }
          }
          style={
            Object {
              "borderRadius": 18,
              "height": 36,
              "width": 36,
            }
          }
          virtual={undefined}
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "alignSelf": "center",
          "flex": 0.8,
        }
      }
      virtual={undefined}
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "paddingTop": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": null,
              "color": "#FF5722",
              "fontFamily": "System",
              "fontSize": 12,
            }
          }
          virtual={undefined}
        >
          CANCELLED
           
        </Text>
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Object {
                "color": "#FF5722",
                "fontSize": 17,
                "paddingRight": 5,
              },
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          Ôçß
        </Text>
      </View>
    </View>
  </View>
</View>
`;
